controllers/project_reg.go

```diff
@@ -0,0 +1,145 @@
+package controllers
+
+import (
+	"fmt"
+	"kwoc-backend/middleware"
+	"kwoc-backend/models"
+	"kwoc-backend/utils"
+	"net/http"
+
+	"github.com/rs/zerolog/log"
+	"gorm.io/gorm"
+)
+
+type RegisterProjectReqFields struct {
+	// Name of the project
+	Name string `json:"name"`
+	// Description for the project
+	Description string `json:"desc"`
+	// List of tags for the project
+	Tags string `json:"tags"`
+	// Mentor's username
+	MentorUsername string `json:"username"`
+	// Secondary mentor's username
+	SecondaryMentorUsername string `json:"secondaryMentor"`
+	// Link to the repository of the project
+	RepoLink string `json:"repoLink"`
+	// Link to a communication channel/platform
+	ComChannel string `json:"comChannel"`
+	// Link to the project's README file
+	ReadmeURL string `json:"readme"`
+}
+
+func RegisterProject(w http.ResponseWriter, r *http.Request) {
+	app := r.Context().Value(middleware.APP_CTX_KEY).(*middleware.App)
+	db := app.Db
+
+	reqFields := RegisterProjectReqFields{}
+
+	err := utils.DecodeJSONBody(r, &reqFields)
+
+	if err != nil {
+		utils.LogErrAndRespond(r, w, err, "Error decoding request JSON body.", http.StatusBadRequest)
+		return
+	}
+
+	login_username := r.Context().Value(middleware.LoginCtxKey(middleware.LOGIN_CTX_USERNAME_KEY))
+
+	if reqFields.MentorUsername != login_username {
+		log.Warn().Msgf(
+			"%s %s %s\n%s %s",
+			r.Method,
+			r.RequestURI,
+			"POSSIBLE SESSION HIJACKING.",
+			fmt.Sprintf("JWT Username: %s", login_username),
+			fmt.Sprintf("Given Username: %s", reqFields.MentorUsername),
+		)
+
+		utils.LogWarnAndRespond(r, w, "Login username and mentor username do not match.", http.StatusUnauthorized)
```

Comment:
```suggestion
	utils.LogWarnAndRespond(r, w, "Login username and mentor username do not match.", http.StatusUnauthorized)
```

Ig the function logs too